kafka/src/main/java/com/metamx/tranquility/kafka/writer/TranquilityEventWriter.java

```diff
@@ -0,0 +1,161 @@
+ * regarding copyright ownership.  Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.metamx.tranquility.kafka.writer;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Throwables;
+import com.google.common.primitives.Ints;
+import com.metamx.common.logger.Logger;
+import com.metamx.common.scala.net.curator.Disco;
+import com.metamx.tranquility.config.ConfigHelper;
+import com.metamx.tranquility.config.DataSourceConfig;
+import com.metamx.tranquility.druid.DruidLocation;
+import com.metamx.tranquility.finagle.FinagleRegistry;
+import com.metamx.tranquility.finagle.FinagleRegistryConfig;
+import com.metamx.tranquility.kafka.model.MessageCounters;
+import com.metamx.tranquility.kafka.model.TranquilityKafkaConfig;
+import com.metamx.tranquility.tranquilizer.SimpleTranquilizerAdapter;
+import com.metamx.tranquility.tranquilizer.Tranquilizer;
+import org.apache.curator.RetryPolicy;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.retry.ExponentialBackoffRetry;
+import scala.Predef;
+import scala.Tuple2;
+import scala.collection.JavaConverters;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Pushes events to Druid through Tranquility using the SimpleTranquilizerAdapter.
+ */
+public class TranquilityEventWriter
+{
+  private static final Logger log = new Logger(TranquilityEventWriter.class);
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+  private static final RetryPolicy RETRY_POLICY = new ExponentialBackoffRetry(1000, 500, 30000);
+
+  private final CuratorFramework curator;
+  private final DataSourceConfig<TranquilityKafkaConfig> dataSourceConfig;
+  private final SimpleTranquilizerAdapter<scala.collection.immutable.Map<String, Object>> simpleTranquilizerAdapter;
+
+  private long rejectedLogCount = 0;
+
+  public TranquilityEventWriter(String topic, DataSourceConfig<TranquilityKafkaConfig> dataSourceConfig)
+  {
+    this.dataSourceConfig = dataSourceConfig;
+
+    final int zkTimeout = Ints.checkedCast(
+        dataSourceConfig.config()
+                        .zookeeperTimeout()
+                        .toStandardDuration()
+                        .getMillis()
+    );
+
+    curator = CuratorFrameworkFactory.builder()
```

Comment:
Curator and FinagleRegistry instances should be shared amongst all TranquilityEventWriters, as each one maintains threads and connections-- see how ServerMain handles that for an example.